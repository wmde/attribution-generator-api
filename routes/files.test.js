const setup = require('./__helpers__/setup');

describe('files routes', () => {
  let context;

  beforeEach(async () => {
    context = await setup({});
  });

  afterEach(async () => {
    await context.destroy();
  });

  describe('GET /files', () => {
    function options() {
      return { url: '/files', method: 'GET' };
    }

    async function subject() {
      return context.inject(options());
    }

    it('returns a list of files with their name and url', async () => {
      const response = await subject({});
      expect(response.status).toBe(200);
      expect(response.type).toBe('application/json');
      expect(response.payload).toMatchSnapshot();
    });
  });
});
