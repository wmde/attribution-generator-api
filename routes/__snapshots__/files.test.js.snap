// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`files routes GET /files returns 422 response if the URL is invalid 1`] = `
Object {
  "error": "Unprocessable Entity",
  "message": "invalid-url",
  "statusCode": 422,
}
`;

exports[`files routes GET /files returns a 404 when called with an unencoded articleUrl 1`] = `
Object {
  "error": "Not Found",
  "message": "Not Found",
  "statusCode": 404,
}
`;

exports[`files routes GET /files returns a 500 response for a generic error 1`] = `
Object {
  "error": "Internal Server Error",
  "message": "An internal server error occurred",
  "statusCode": 500,
}
`;

exports[`files routes GET /files returns a 503 response when the wiki api is not reachable 1`] = `
Object {
  "error": "Service Unavailable",
  "message": "api-unavailable",
  "statusCode": 503,
}
`;

exports[`files routes GET /files returns a list of all files from the given article 1`] = `
Array [
  Object {
    "file": "File:image.jpg",
    "url": "https://en.wikipedia.org/wiki/File:image.jpg",
  },
]
`;
